plugins/inputs/execd/shim/shim_test.go

```diff
@@ -172,3 +160,15 @@ func (i *serviceInput) Start(acc telegraf.Accumulator) error {
 
 func (i *serviceInput) Stop() {
 }
+
+// we can get stuck if stdout gets clogged up and nobody's reading from it.
+// make sure we keep it going
+func readUntilEmpty(r *bufio.Reader) {
```

Comment:
Shouldn't the shim code we are testing be doing this?

Reply:
no, because this is the read end of the out pipe coming from the shim.